### PR TITLE
feat: user group membership identity unified api

### PR DESCRIPTION
**Pull request by ghost**
## Description
* Unified API endpoints are available for 
  * add user to group
  * remove user from group
  * find all users of group

## Related issues

closes https://github.com/camunda-cloud/identity/issues/2863

### CODE REVIEW

File: zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/ErrorMapperTest.java [hunk 1]
```diff
@@ -286,6 +287,11 @@ public ProcessInstanceServices processInstanceService() {
       return Mockito.mock(ProcessInstanceServices.class);
     }
 
+    @Bean
+    public GroupController groupController() {
+      return Mockito.mock(GroupController.class);
+    }
+
```
Comment (tmetzke) [inline zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/ErrorMapperTest.java:294]: ❌ Adjusting this class will become obsolete soon after https://github.com/camunda/camunda/pull/19978 is merged 👍 🙂

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -319,7 +319,7 @@ paths:
               schema:
                 $ref: "#/components/schemas/ProblemDetail"
         "404":
-          description: "Not found"
+          description: "Bad request"
```
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:322]: ❌ We must not override this back to the previous values. The 404 code is clearly a "Not found".

File: zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/GroupControllerTest.java [hunk 3]
```diff
@@ -31,6 +32,7 @@
 public class GroupControllerTest extends RestControllerTest {
```
Comment (tmetzke) [inline zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/GroupControllerTest.java:32]: ❌ Let's add tests to this for membership requests 👍

**Review by tmetzke — CHANGES_REQUESTED** [review]
👍 Good start to this issue
🔧 I added a couple of suggestions and improvement ideas.

**Review by tmetzke — CHANGES_REQUESTED** [review]
👍 Really good progress, @maryarm 🚀 
🔧 I added a couple of suggestions, nothing major though

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java [hunk 4]
```diff
@@ -247,6 +254,42 @@ final var record = new UserTaskRecord();
     return record;
   }
 
+  public static CamundaUserWithPassword getUserWithPassword(
```
[the remark this answers is not in the document]
Comment (maryarm) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java:257]: Ah, Ok. I'll change. but the reason that I used get was some other methods that are already there is started with `get` not `to` like `getRecordWithChangedAttributes`. I though it's the standard.

[the remark this answers is not in the document]
Comment (tmetzke) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java:257]: I see what you mean, yeah, you're right. You can rename that one method to `toUserTaskRecordWithChangedAttributes` actually. I wasn't really precise with that one back in the days, good catch 👍 Let's always directly improve what we can improve 🙂 
The other `get...` methods can stay as they are, I think. They are actually getting things out of existing data mostly.

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/ResponseMapper.java [hunk 5]
```diff
@@ -56,6 +60,24 @@ private static ActivatedJob toActivatedJob(final long jobKey, final JobRecord jo
...
+
+    return camundaUserDto;
+  }
+
+  public static CamundaGroupResponse toGroupResponse(final CamundaGroup group) {
+    final CamundaGroupResponse camundaGroupResponse = new CamundaGroupResponse();
+    camundaGroupResponse.setId(group.id());
+    camundaGroupResponse.setName(group.name());
+    return camundaGroupResponse;
+  }
```
Comment (tmetzke) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/ResponseMapper.java:79]: 🙃 Just a nitpick: let's align the names. Either both are `toCamunda...Response` or both are `to...Response`.

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (maryarm) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java]: I had another look and I decided not to change `getRecordWithChangedAttributes` as it is a private method used only internally. So i assume the `to` prefix is for public mapper methods. :)

**Comment by tmetzke** [comment]
For future reference (you don't have to change this now): if you adjust code but don't fix a bug/defect, you can use the `refactor` type in the commit message. That way, it's easy to see what is an internal adjustment without a behavioral change.

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (tmetzke) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java]: That also makes sense, yeah, I agree 👍 Let's keep the private method as is then 🙂

**Review by tmetzke — APPROVED** [review]
👍 Really well done ⭐ 
Thanks for swiftly looking into the suggestions and addressing them 🚀

**Comment by maryarm** [comment]
I have to close this PR and create a new one to solve the ghost account problem during merge.

**Pull request closed without merging**